packages/shared/src/hooks/useBoot.ts

```diff
@@ -0,0 +1,27 @@
+import { useQueryClient } from 'react-query';
+import { BOOT_QUERY_KEY } from '../contexts/common';
+import { squadMemberships } from '../graphql/squads';
+import { Boot } from '../lib/boot';
+
+type UseBoot = {
+  deleteSquad: (squadId: string) => void;
+  updateSquads: () => Promise<void>;
+};
+
+export const useBoot = (): UseBoot => {
+  const client = useQueryClient();
+  const deleteSquad = (squadId: string) => {
+    const bootData = client.getQueryData<Boot>(BOOT_QUERY_KEY);
+    const squads = bootData.squads?.filter((squad) => squad.id !== squadId);
+    client.setQueryData<Boot>(BOOT_QUERY_KEY, { ...bootData, squads });
+  };
+  const updateSquads = async () => {
+    const squads = await squadMemberships();
```

Comment:
For this one, can't we also simply push the squad we just got in return from the create callback?

Reply:
What will be the order? Is it going to be at the end?

Reply:
I'd say where you get the response from `createSquad` (which includes the squad) we can simply call `updateSquads` and pass that single squad right?

Let me know if I overlooked something.

Reply:
No, I meant once we have the squad on our hand, what will be its position on the squad list. I think this is ordered by something:
<img width="50" alt="image" src="https://user-images.githubusercontent.com/1488164/214526065-0e11a203-c293-4f8e-a143-6ddb3f0da4f1.png">

Reply:
ok, looks like it is ordered by squad name

Reply:
Oh sorry yes by name `.orderBy('s.name', 'ASC')`